addOns/automation/src/main/java/org/zaproxy/addon/automation/AutomationAPI.java

```diff
@@ -0,0 +1,148 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.addon.automation;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.HashMap;
+import net.sf.json.JSONObject;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.parosproxy.paros.view.View;
+import org.zaproxy.zap.extension.api.ApiAction;
+import org.zaproxy.zap.extension.api.ApiException;
+import org.zaproxy.zap.extension.api.ApiException.Type;
+import org.zaproxy.zap.extension.api.ApiImplementor;
+import org.zaproxy.zap.extension.api.ApiResponse;
+import org.zaproxy.zap.extension.api.ApiResponseElement;
+import org.zaproxy.zap.extension.api.ApiResponseList;
+import org.zaproxy.zap.extension.api.ApiResponseSet;
+import org.zaproxy.zap.extension.api.ApiView;
+import org.zaproxy.zap.utils.ApiUtils;
+
+public class AutomationAPI extends ApiImplementor {
+
+    private static final Logger LOG = LogManager.getLogger(AutomationAPI.class);
+
+    private static final String PREFIX = "automation";
+
+    private static final String ACTION_RUN_PLAN = "runplan";
+    private static final String VIEW_PLAN_PROGRESS = "planprogress";
+
+    private static final String PARAM_FILE_PATH = "filePath";
+    private static final String PARAM_PLAN_ID = "planId";
+
+    private static final String ELEMENT_INFO = "info";
+    private static final String ELEMENT_WARN = "warn";
+    private static final String ELEMENT_ERROR = "error";
+
+    private ExtensionAutomation extension;
+
+    /** Provided only for API client generator usage. */
+    public AutomationAPI() {
+        this(null);
+    }
+
+    public AutomationAPI(ExtensionAutomation extension) {
+        super();
+        this.extension = extension;
+        this.addApiAction(new ApiAction(ACTION_RUN_PLAN, new String[] {PARAM_FILE_PATH}));
+        this.addApiView(new ApiView(VIEW_PLAN_PROGRESS, new String[] {PARAM_PLAN_ID}));
+    }
+
+    @Override
+    public String getPrefix() {
+        return PREFIX;
+    }
+
+    @Override
+    public ApiResponse handleApiAction(String name, JSONObject params) throws ApiException {
+        LOG.debug("handleApiAction {} {}", name, params);
+
+        if (name.equals(ACTION_RUN_PLAN)) {
+            try {
+                AutomationPlan plan =
+                        extension.loadPlan(
+                                new File(ApiUtils.getNonEmptyStringParam(params, PARAM_FILE_PATH)));
+                extension.registerPlan(plan);
+
+                if (View.isInitialised()) {
+                    extension.displayPlan(plan);
+                }
+                extension.runPlanAsync(plan);
+
+                return new ApiResponseElement(PARAM_PLAN_ID, Integer.toString(plan.getId()));
+            } catch (IOException | ApiException e) {
+                throw new ApiException(Type.DOES_NOT_EXIST, e.getMessage());
+            }
+        } else {
```

Comment:
The else could be removed (the then either throws or returns), same comment for the view handling.